Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -103,6 +103,21 @@ class ScoreMatchesByDegreeOfCoreSubstitution {
 };
 }  // namespace detail
 
+bool propertyCompat(const RDProps *r1, const RDProps *r2,
+                    const std::vector<std::string>& properties) {
+  for (const auto &prop : properties) {
+    if (r1->hasProp(prop) && r2->hasProp(prop)) {
```

Comment:
This might be more efficient as:

```
std::string prop1;
std::string prop2;
bool hasprop1  = r1->getPropIfPresent<std::string>(prop, prop1);
bool hasprop2 =  hasprop1 && r2->getPropIfPresent<std::string>(prop, prop2);
if(hasprop1 && hasprop2) {
  return prop1 == prop2;
} 
return !(hasprop1 || hasprop2);
}
```

Reply:
I think my logic is right here.

Reply:
I think that would works for a single property but I think we want to check all of them , but I will update to use `getPropIfPresent`